From 1b735107e80cde5b2b22ce1fedb2e329961c6459 Mon Sep 17 00:00:00 2001
From: Rahul Roshan Kachchap <rahulroshan.kachchap@windriver.com>
Date: Fri, 4 Oct 2024 04:00:07 -0400
Subject: [PATCH] Improving delete error message for keystone user accounts

Signed-off-by: Rahul Roshan Kachchap <rahulroshan.kachchap@windriver.com>
---
 keystone/exception.py | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/keystone/exception.py b/keystone/exception.py
index c62338b89..cacf1586b 100644
--- a/keystone/exception.py
+++ b/keystone/exception.py
@@ -350,15 +350,17 @@ class AdditionalAuthRequired(AuthPluginException):
 
 
 class Forbidden(SecurityError):
-    message_format = _("You are not authorized to perform the"
-                       " requested action.")
+    message_format = _("You are forbidden to perform the requested action. "
+                       "This action is system-critical and cannot be executed- %(action)s. "
+                       "Please contact your administrator for further assistance.")
     code = int(http.client.FORBIDDEN)
     title = http.client.responses[http.client.FORBIDDEN]
 
 
 class ForbiddenAction(Forbidden):
-    message_format = _("You are not authorized to perform the"
-                       " requested action: %(action)s.")
+    message_format = _("You are forbidden to perform the requested action. "
+                       "This action is system-critical and cannot be executed- %(action)s. "
+                       "Please contact your administrator for further assistance.")
 
 
 class CrossBackendNotAllowed(Forbidden):
-- 
2.25.1

