From 7feac57d571e49e042adb96738a3688c56adade0 Mon Sep 17 00:00:00 2001
From: Tyler Smith <tyler.smith@windriver.com>
Date: Mon, 8 Apr 2019 15:33:16 -0400
Subject: [PATCH 1/1] Add keyring patch

Signed-off-by: Yong Fu <fuyong@neusoft.com>
---
 SPECS/openstack-keystone.spec | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/SPECS/openstack-keystone.spec b/SPECS/openstack-keystone.spec
index d3ec1d7..3d5fcee 100644
--- a/SPECS/openstack-keystone.spec
+++ b/SPECS/openstack-keystone.spec
@@ -27,7 +27,7 @@ Name:           openstack-keystone
 # https://review.openstack.org/#/q/I6a35fa0dda798fad93b804d00a46af80f08d475c,n,z
 Epoch:          1
 Version:        16.0.0
-Release:        1%{?dist}
+Release:        1.el8%{?_tis_dist}.%{tis_patch_ver}
 Summary:        OpenStack Identity Service
 License:        ASL 2.0
 URL:            http://keystone.openstack.org/
@@ -39,6 +39,9 @@ Source3:        openstack-keystone.sysctl
 Source5:        openstack-keystone-sample-data
 Source20:       keystone-dist.conf
 
+# STX: Include patches here
+Patch1:         0001-Rebasing-Keyring-integration.patch
+
 BuildArch:      noarch
 BuildRequires:  openstack-macros
 BuildRequires:  python%{pyver}-devel
-- 
2.17.1

