From 9eb2a001b8f80d1a66d4ec78d882d7d6cb4dff51 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Tue, 24 Jan 2017 12:16:38 -0500
Subject: [PATCH 1/2] WRS: 0001-Update-package-versioning-for-TIS-format.patch

Signed-off-by: zhipengl <zhipengs.liu@intel.com>
---
 SPECS/rabbitmq-server.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/rabbitmq-server.spec b/SPECS/rabbitmq-server.spec
index db01723..7b879ed 100644
--- a/SPECS/rabbitmq-server.spec
+++ b/SPECS/rabbitmq-server.spec
@@ -7,7 +7,7 @@
 
 Name: rabbitmq-server
 Version: 3.8.2
-Release: 3%{?dist}
+Release: 3.el8%{?_tis_dist}.%{tis_patch_ver}
 License: MPLv1.1
 Source0: https://github.com/rabbitmq/rabbitmq-server/releases/download/v%{version}/%{name}_%{version}.orig.tar.xz
 Source1: https://github.com/rabbitmq/rabbitmq-server/releases/download/v%{version}/%{name}_%{version}.orig.tar.xz.asc
-- 
2.7.4

