From e7c67702c374d330ca99cf25ff09902228ce1924 Mon Sep 17 00:00:00 2001
From: Long Li <lilong-neu@neusoft.com>
Date: Fri, 15 Nov 2019 16:39:58 +0800
Subject: [PATCH] upgrade_new_official_way_get_stack_trace

---
 SPECS/rabbitmq-server.spec | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/SPECS/rabbitmq-server.spec b/SPECS/rabbitmq-server.spec
index 3a7d940..47e7748 100644
--- a/SPECS/rabbitmq-server.spec
+++ b/SPECS/rabbitmq-server.spec
@@ -22,12 +22,15 @@ Patch3: rabbitmq-server-0003-Revert-Distinct-exit-codes-for-CLI-utilities.patch
 Patch4: rabbitmq-server-0004-Allow-guest-login-from-non-loopback-connections.patch
 Patch5: rabbitmq-server-0005-Avoid-RPC-roundtrips-in-list-commands.patch
 Patch6: rabbitmq-server-0006-rabbit_prelaunch-must-use-RABBITMQ_SERVER_ERL_ARGS.patch
+
+Patch7: rabbitmq-server-0007-upgrade_new_official_way_to_get_stacktrace.patch
+
 Patch101: rabbitmq-common-0001-Avoid-RPC-roundtrips-while-listing-items.patch
 Patch102: rabbitmq-common-0002-Use-proto_dist-from-command-line.patch
 
 URL: http://www.rabbitmq.com/
 BuildArch: noarch
-BuildRequires: erlang >= %{erlang_minver}, python-simplejson, xmlto, libxslt, python, zip
+BuildRequires: erlang >= %{erlang_minver}, python3-simplejson, xmlto, libxslt, python3, zip
 BuildRequires: rsync
 BuildRequires: systemd
 Requires: logrotate
@@ -63,6 +66,7 @@ scalable implementation of an AMQP broker.
 %patch4 -p1
 %patch5 -p1
 %patch6 -p1
+%patch7 -p1
 
 cd deps/rabbit_common
 %patch101 -p1
-- 
2.7.4

